addons/account/account_move_line.py

```diff
@@ -1057,14 +1063,18 @@ def reconcile(self, cr, uid, ids, type='auto', writeoff_acc_id=False, writeoff_p
                     'amount_currency': amount_currency_writeoff and amount_currency_writeoff or (account.currency_id.id and currency or 0.0)
                 })
             ]
-
-            writeoff_move_id = move_obj.create(cr, uid, {
-                'period_id': writeoff_period_id,
-                'journal_id': writeoff_journal_id,
-                'date':date,
-                'state': 'draft',
-                'line_id': writeoff_lines
-            })
+            if context.get('bs_move_id'):
+                writeoff_move_id = context['bs_move_id']
+                for l in writeoff_lines:
+                    self.create(cr, uid, dict(l[2], move_id=writeoff_move_id), context=context)
```

Comment:
don't you need to add something in the context to have not raising error about "unbalanced move"?